File: misc/samples/find_pattern.pl
use warnings;
use Time::HiRes qw/time/;

my $start=time;

open(FD, "c:/yunshang_10m_line.log") || die("can not open file");
my $num = 0;
my $total = 0;
while(<FD>){
  $total++;
  if(m/cur_wnd.*(20|21|31)/){
    $num++;
  }
}
print("find ".$num." match \n");
print("total ".$total." lines \n");
my $end = time;
print("cost ".($end-$start)." sec \n");

File: misc/samples/find_pattern.py
# coding=utf-8
# author: zengyuetian


import time
import re
t1 = time.time()
total = 0
num = 0
pattern = re.compile("cur_wnd.*(20|21|31)")
with open("c:/yunshang_10m_line.log", "r") as f:
    line = f.readline()
    while line:
        total += 1
        if pattern.search(line):
            num += 1
        line = f.readline()
print "find {0} match".format(num)
print "total {0} lines".format(total)
t2 = time.time()
print "cost {0} sec".format(t2-t1)




